scene/cameras.py

```diff
@@ -36,6 +36,9 @@ def __init__(self, colmap_id, R, T, FoVx, FoVy, image, gt_alpha_mask,
             print(f"[Warning] Custom device {data_device} failed, fallback to default cuda device" )
             self.data_device = torch.device("cuda")
 
+        if lazy_load:
+            self.data_device = torch.device("cpu")
```

Comment:
Maybe equivalent?

```
if lazy_load:
    data_device = "cpu"

try:
    self.data_device = torch.device(data_device)
....


If this is correct, perhaps `lazy_load` can be replaced directly with `data_device` (and it needs to be exposed in `train.py`). Perhaps adding a comment in the README, there is a question about the memory and the different approaches when there is no 24GB.

Reply:
After your reminder, I realized that other tensors were directly migrated through `. cuda`, and `data_device` is only responsible for images, which can indeed function as a lazy load.

There is no need to expose `data_device` in `train.py` as it is already specified in the `arguments/__init__.py`.

All in all, nothing needs to be modified. If possible, I suggest modifying the "24 GB VRAM" in README, as it can easily make people mistakenly think that this is the minimum configuration.

Reply:
I am not sure to understand your reply. What do you mean by `. cuda`? I think the lazy_load was a good option to add, so you don't need to modify every time the `__init__.py`. Perhaps, could you expand how you would modify the README to clarify that `data_device` can be used for lazy loading?

Reply:
"other tensors were directly migrated through . cuda" precisely means that they are fixed to the cuda device, e.g. [create gaussian](https://github.com/graphdeco-inria/gaussian-splatting/blob/main/scene/gaussian_model.py#L124-L147).

There is no need to need to modify `data_device` every time the `__init__.py`. The train.py uses [`argparse.ArgumentParser`](https://github.com/graphdeco-inria/gaussian-splatting/blob/main/train.py#L195-L208) to analyse the command arguments. And [`data_device`](https://github.com/graphdeco-inria/gaussian-splatting/blob/main/arguments/__init__.py#L55) is can be called by `python train.py --data_device cpu -s <path to COLMAP or NeRF Synthetic dataset>`.

```python
# scene/cameras.py
# if here data_device is specified as `cpu`, self.original_image is on cpu
try:
    self.data_device = torch.device(data_device)
except Exception as e:
    print(e)
    print(f"[Warning] Custom device {data_device} failed, fallback to default cuda device" )
    self.data_device = torch.device("cuda")

self.original_image = image.clamp(0.0, 1.0).to(self.data_device)
self.image_width = self.original_image.shape[2]
self.image_height = self.original_image.shape[1]

if gt_alpha_mask is not None:
    self.original_image *= gt_alpha_mask.to(self.data_device)
else:
    self.original_image *= torch.ones((1, self.image_height, self.image_width), device=self.data_device)

# train.py
# original_image is sent to gpu while iterating
gt_image = viewpoint_cam.original_image.cuda()
```

![image](https://github.com/graphdeco-inria/gaussian-splatting/assets/32301361/a4891a45-d0f3-4526-9fc8-1962727dff63)

[`data_device` in README](https://github.com/graphdeco-inria/gaussian-splatting/blob/2eee0e26d2d5fd00ec462df47752223952f6bf4e/README.md?plain=1#L134)

Reply:
I modify REAMDE to clarify lazy load and recommit it.

Reply:
thanks for the clarifications!